lib/pbench/agent/tool_meister.py

```diff
@@ -200,210 +347,141 @@ def stop(self):
-            self.start_process.kill()
-            self.logger.warning(
-                "Killed un-responsive transient tool %s start process after"
-                " 30 seconds",
-                self.name,
-            )
-        else:
-            if sts != 0 and sts != -(signal.SIGTERM):
-                self.logger.warning(
-                    "Transient tool %s start process failed with %d", self.name, sts
-                )
-        self.start_process = None
+        # First wait for the "stop" process to do it's job ...
+        self.stop_process = self._wait_for_process_with_kill(self.stop_process, "stop")
+        # ... then we wait for the start process to finish; in either case,
+        # we'll only wait a short time before killing them.
+        self.start_process = self._wait_for_process_with_kill(
+            self.start_process, "start"
+        )
 
 
-class PcpTransTool(Tool):
-    """PcpTransTool - A more traditional alternative to the pcp persistent tool that
-                      allows one to run both the pmlogger and pmcd locally on each
-                      registered node. Additionally, this tool starts, stops, and
-                      sends data alongside other transient tools, rather than always
-                      running in the background.
+class PcpTransientTool(Tool):
+    """The transient tool alternative to the PCP persistent tool, which starts
+    and stops both a pmcd and pmlogger process, and sends data remotely, as
+    directed for transient tools.
     """
 
-    def __init__(self, name, tool_opts, logger=None, **kwargs):
-        super().__init__(name, tool_opts, logger=logger, **kwargs)
-        if self.tool_dir:
-            self.tool_dir = self.tool_dir / self.name.replace("-transient", "")
-        if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
-            os.environ["PATH"] += os.pathsep + "/usr/libexec/pcp/bin"
-        pmcd_path = find_executable("pmcd")
-        if pmcd_path:
-            self.pmcd_args = [
-                pmcd_path,
-                "--foreground",
-                "--socket=./pmcd.socket",
-                "--port=55677",
-                f"--config={self.pbench_install_dir}/templates/pmcd.conf",
-            ]
-        else:
-            self.pmcd_args = None
-        pmlogger_path = find_executable("pmlogger")
-        if pmlogger_path:
-            self.pmlogger_args = [
-                pmlogger_path,
-                "--log=-",
-                "--report",
-                "-t",
-                "3s",
-                "-c",
-                f"{self.pbench_install_dir}/templates/pmlogger.conf",
-                "--host=localhost:55677",
-                f"{self.tool_dir}/%Y%m%d.%H.%M",
-            ]
-        else:
-            self.pmlogger_args = None
+    def __init__(self, name, tool_opts, **kwargs):
+        super().__init__(name, tool_opts, **kwargs)
+        self.real_name = self.name.replace("-transient", "")
```

Comment:
`real_name` is an odd name for this field....especially since this field is almost never used, relative to `name`.

Since it's only used in one place, to generate a subdirectory name, it strikes me that maybe the directory name should be generated here, instead.  Or, at least, this field should be named something more like `subdir_name`.